pipelinewise/fastsync/commons/transform_utils.py

```diff
@@ -0,0 +1,278 @@
+
+                    trans_map.append({
+                        'trans': cls.__hash_skip_first_to_sql(transform_type, column, sql_flavor),
+                        'conditions': conditions
+                    })
+
+                elif transform_type == TransformationType.MASK_DATE:
+
+                    trans_map.append({
+                        'trans': cls.__mask_date_to_sql(column, sql_flavor),
+                        'conditions': conditions
+                    })
+
+                elif transform_type == TransformationType.MASK_NUMBER:
+                    trans_map.append({
+                        'trans': f'{column} = 0',
+                        'conditions': conditions
+                    })
+
+                elif transform_type == TransformationType.MASK_HIDDEN:
+                    trans_map.append({
+                        'trans': f"{column} = 'hidden'",
+                        'conditions': conditions
+                    })
+
+        return trans_map
+
+    @classmethod
+    def __conditions_to_sql(
+            cls,
+            transform_conditions: List[Dict],
+            sql_flavor: SQLFlavor) -> Optional[str]:
+        """
+        Convert the conditional transformations into equivalent form in SF SQL.
+        Args:
+            transform_conditions: A list of dictionaries of the either form
+                {
+                    "column": "...",        # required
+                    "safe_column": "...",   # optional
+                    "equals": <>
+                }
+
+                {
+                    "column": "...",        # required
+                    "safe_column": "...",   # optional
+                    "regex_match": "..."
+                }
+
+                if no regex_match or equals keys are found, the transformation condition is skipped
+
+        Returns: None if no transformations, otherwise a concatenated string of AND conditions
+        """
+        if not transform_conditions:
+            return None
+
+        conditions = []
+
+        for condition in transform_conditions:
+            # for each condition create the sql equivalent of it
+            if 'equals' in condition:
+                if condition['equals'] is None:
+                    operator, value = 'IS', 'NULL'
+
+                elif condition['equals'] == '':
+                    operator, value = '=', "''"
+
+                else:
+                    operator = '='
+                    value = f"'{condition['equals']}'" if isinstance(condition['equals'], str) else condition['equals']
+
+            elif 'regex_match' in condition:
+
+                if sql_flavor == SQLFlavor.SNOWFLAKE:
+                    operator = 'REGEXP'
+
+                elif sql_flavor == SQLFlavor.POSTGRES:
+                    operator = '~'
+
+                else:
+                    raise NotImplementedError(f'regex_match conditional transformation in {sql_flavor.value} SQL '
```

Comment:
what about the redshift target? Is that still working as of today, or it will raise this exception?

Reply:
redshift target is not using this `TransformationHelper` so if it's working now, it will keep on working.